src/BridgedERC20.sol

```diff
@@ -128,6 +129,9 @@ contract BridgedERC20 is ERC20, AccessControlDefaultAdminRules {
         // Restrictions ignored for minting and burning
         // If transferRestrictor is not set, no restrictions are applied
         if (from == address(0) || to == address(0) || address(transferRestrictor) == address(0)) {
+            if (to == address(0)) {
+                if (!hasRole(BURNER_ROLE, msg.sender)) revert UnauthorizedOperation();
```

Comment:
I would rather check against burn call than role. That way, the role doesn't override the check behavior. Check msg.sig == burn.selector